### PR TITLE
Tabbed Analytics Dashboard View

### PR DESCRIPTION
**Pull request by ibnc**
Adding the Global and Pipeline tabs to the analytics dashboard

Global Tab:
global_tab

Pipelines Tab (1):
pipeline_tab_1

Pipelines Tab (2):
pipeline_tab_2

### CODE REVIEW

**Comment by ibnc** [comment]
@maheshp @jyotisingh @arvindsv Would one of you mind taking a look at this?

**Comment by maheshp** [comment]
@ibnc the position of the tabs changes between selected tabs.

On Global tab
screen shot 2018-03-09 at 1 50 41 pm

On Pipeline tab
screen shot 2018-03-09 at 1 50 56 pm

**Comment by maheshp** [comment]
The drop width seems too wide, what do you think? @tinozza 

screen shot 2018-03-09 at 1 53 45 pm

**Comment by maheshp** [comment]
Navigating between tabs does not retain the selected pipeline in dropdown.

Steps to reproduce.
1. Create pipeline P1 and P2 and trigger couple of runs for each.
2. Go to top level analytics page
3. On pipeline tab select P2 pipeline in dropdown
4. Now open the Global tab
5. Come back to pipeline tab
You will see graph for P2 displayed, but the dropdown would show P1 selected.

File: server/webapp/WEB-INF/rails.new/webpack/single_page_apps/analytics.js [hunk 1]
```diff
@@ -25,64 +25,28 @@
...
-
-            frames.push(m(PluginiFrameWidget, {model, pluginId, uid, init: PluginEndpoint.init}));
-          });
-        });
-        return frames;
+        const metrics = $(main).data("supported-dashboard-metrics");
+        const pageItems = [];
+        const tabs = new Tabs(m.redraw);
+        pageItems.push(m(AnalyticsDashboardHeader));
+        tabs.push(new Tab("Global", GlobalMetrics, metrics));
```
Comment (maheshp) [inline server/webapp/WEB-INF/rails.new/webpack/single_page_apps/analytics.js]: We are using foundation tabs in few places, there is a form_helper which exposes the tabs https://github.com/gocd/gocd/blob/master/server/webapp/WEB-INF/rails.new/webpack/helpers/form_helper.js.msx#L271 . 

Please check if it simplifies things, you can refer to its usage - https://github.com/gocd/gocd/blob/master/server/webapp/WEB-INF/rails.new/webpack/views/pipeline_configs/jobs_config_widget.js.msx#L333

File: server/webapp/WEB-INF/rails.new/webpack/models/analytics/tabs.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+(function() {
+  "use strict";
+
+  function Tabs(callback) {
```
[the remark this answers is not in the document]
Comment (ibnc) [reply server/webapp/WEB-INF/rails.new/webpack/models/analytics/tabs.js]: In what respect is this model different? The only difference I can see is that Tabs isn't using Stream (for which I see no reasons why Tabs would need to use Stream).

File: server/webapp/WEB-INF/rails.new/webpack/single_page_apps/analytics.js [hunk 3]
```diff
@@ -25,64 +25,28 @@
...
-            const uid = `f-${pluginId}:${metric}:${idx}`,
-              model = ensureModel(uid, pluginId, metric);
-
-            frames.push(m(PluginiFrameWidget, {model, pluginId, uid, init: PluginEndpoint.init}));
-          });
-        });
-        return frames;
+        const metrics = $(main).data("supported-dashboard-metrics");
+        const pageItems = [];
+        const tabs = new Tabs(m.redraw);
```
[the remark this answers is not in the document]
Comment (ibnc) [reply server/webapp/WEB-INF/rails.new/webpack/single_page_apps/analytics.js]: I'm not sure I understand how `new Tabs(m.redraw)` is different from how Frames are initialized (`new Frame(m.redraw)`)...

**Comment by ibnc** [comment]
@maheshp I pushed a fix for the pipeline dropdown not remembering which pipeline was selected

**Pull request merged**